### DISCUSSION TITLE
Cannot use the accessibility ID on Android with React Native?

### DISCUSSION THREAD

**Discussion by mtaylor-evidation** [webdriverio/webdriverio#8761] [Technical Discussion]
I'm struggling to find a good solution to utilizing the accessibility ID on Android. In the webdriverio documentation this appears to be the ideal element selector option. However, when I try and use it exactly the same way I do on iOS it fails for Android (not it works with iOS perfectly).

Here are my desired capabilities settings for android:

```
  capabilities: {
    platformName: "android",
    automationName: "Appium",
    platformVersion: "12",
    deviceName: "emulator-5554",
    app: "path-to-my-apk",
    automationName: "UiAutomator2",
    autoGrantPermissions: 'true',
  }
```

For iOS this is as simple as it gets:

`await client.$('~login--EmailInput').waitForExist({ timeout: 10000 });`

For Android the only thing I can get to work is the xpath:

`await client.$('//*[@resource-id="login--EmailInput"]').waitForExist({ timeout: 10000 });`

When I inspect it with android-inspector it finds it perfectly fine:

[https://ibb.co/2FCJGSx](url)

Any help is appreciated.

**Comment by wswebcreation** [comment]
Hi @mtaylor-evidation 

Can you provide the following:
- a screenshot from Appium Inspector
- versions of React Native you're using
- the logs from WDIO
- the Appium logs

That would help debugging this

Tnx

**Reply by mtaylor-evidation** [reply]
- <img width="1223" alt="Screen Shot 2022-08-24 at 8 02 33 PM" src="https://user-images.githubusercontent.com/61215344/186589981-9707319e-0f8f-4ff4-a930-987c698413ca.png">

- 0.67.4 and 0.68 React Native

- [wdioLogs.docx](https://github.com/webdriverio/webdriverio/files/9421901/wdioLogs.docx)


- [appium-server-logs.txt](https://github.com/webdriverio/webdriverio/files/9421906/appium-server-logs.txt)

Let me know if there is anything else.

**Reply by isaac-tribal** [reply]
@mtaylor-evidation You need to use **accessibilityLabel** property  for Android (not enough testID property)

https://appiumpro.com/editions/76-testing-react-native-apps-with-appium
